tiledb/sm/query/reader.cc

```diff
@@ -2990,6 +2993,10 @@ Status Reader::sparse_read() {
   RETURN_NOT_OK(
       copy_attribute_values(UINT64_MAX, result_tiles, result_cell_slabs));
 
+  if (offsets_extra_element_) {
+    RETURN_NOT_OK(add_extra_offset());
+  }
```

Comment:
This needs to go in `Reader::read` for two reasons:
1. To ensure we call this for dense reads as well (see `dense_read`)
2. This routine, `sparse_read`, may be called multiple times before returning the user. This would result in more than one extra element.

Within `Reader::read`, it loops until returning to the user. This needs to be called right before returning the user. In the current `dev`, I think it needs to be called:
https://github.com/TileDB-Inc/TileDB/blob/dev/tiledb/sm/query/reader.cc#L357
https://github.com/TileDB-Inc/TileDB/blob/dev/tiledb/sm/query/reader.cc#L366

Reply:
Right, sorry I was confused for some reason and though that var-sized attributes are only applicable to sparse arrays (!).
Fixed it and adding a UT to cover dense arrays too

Reply:
By the way, I am not sure to be honest how to give a satisfying branch coverage of Reader::read() in my UTs, meaning I will have to trust you on the 2 places to call `add_extra_offset()` because I don't know how to exercise all the different paths through the ifs with real scenarios...